### PR TITLE
Add consul_license resource

### PR DESCRIPTION
**Pull request by remilapeyre**
Closes https://github.com/terraform-providers/terraform-provider-consul/issues/172

### CODE REVIEW

File: consul/resource_consul_license_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package consul
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccConsulLicense_basic(t *testing.T) {
```
Comment (pearkes) [inline consul/resource_consul_license_test.go]: Is it possible to add some kind of conditional test that runs against the enterprise version?

Comment (remilapeyre) [reply consul/resource_consul_license_test.go]: Do you think it's OK if the test look for a `test_license.hcli` in the working dir and run if it finds one?

Comment (pearkes) [reply consul/resource_consul_license_test.go]: Seems reasonable with documentation and logging. Another option is an API query to /v1/agent/self that maybe interrogates the version string or similar?

Comment (remilapeyre) [reply consul/resource_consul_license_test.go]: I did both, some resources like the `consul_network_segment` or the namespace support can be run without actually having a license when using the `hashicorp/consul-enterprise` Docker image.

**Pull request merged**